### ISSUE TITLE
v4.0.1-php73 tagged failed on PHP 7.3.2

### ISSUE THREAD

**Issue by centminmod** [websupport-sk/pecl-memcache#38]
Trying to compile on PHP-FPM 7.3.2 the (memcache v4.0.1-php73 tagged version](https://github.com/websupport-sk/pecl-memcache/tree/v4.0.1-php73) . Prior to latest memcache 4.x commits, i had memcache 3.0.9 dev working on PHP 7.3.2 with the pull patch at https://github.com/websupport-sk/pecl-memcache/pull/30. If i rollback to commit https://github.com/websupport-sk/pecl-memcache/commit/4a9e4ab0d12150805feca3012854de9fd4e5a721 and apply the patch, memcache 3.0.9 compiles fine on PHP 7.3.2

on make this is error I get

```
make -j$(nproc)
/bin/sh /svr-setup/pecl-memcache-4.0.1-php73/libtool --mode=compile ccache gcc -I/usr/local/include/php -I. -I/svr-setup/pecl-memcache-4.0.1-php73 -DPHP_ATOM_INC -I/svr-setup/pecl-memcache-4.0.1-php73/include -I/svr-setup/pecl-memcache-4.0.1-php73/main -I/svr-setup/pecl-memcache-4.0.1-php73 -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /svr-setup/pecl-memcache-4.0.1-php73/php7/memcache.c -o php7/memcache.lo 
/bin/sh /svr-setup/pecl-memcache-4.0.1-php73/libtool --mode=compile ccache gcc -I/usr/local/include/php -I. -I/svr-setup/pecl-memcache-4.0.1-php73 -DPHP_ATOM_INC -I/svr-setup/pecl-memcache-4.0.1-php73/include -I/svr-setup/pecl-memcache-4.0.1-php73/main -I/svr-setup/pecl-memcache-4.0.1-php73 -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /svr-setup/pecl-memcache-4.0.1-php73/php7/memcache_pool.c -o php7/memcache_pool.lo 
mkdir php7/.libs
mkdir php7/.libs
mkdir: cannot create directory 'php7/.libs': File exists
 ccache gcc -I/usr/local/include/php -I. -I/svr-setup/pecl-memcache-4.0.1-php73 -DPHP_ATOM_INC -I/svr-setup/pecl-memcache-4.0.1-php73/include -I/svr-setup/pecl-memcache-4.0.1-php73/main -I/svr-setup/pecl-memcache-4.0.1-php73 -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /svr-setup/pecl-memcache-4.0.1-php73/php7/memcache_pool.c  -fPIC -DPIC -o php7/.libs/memcache_pool.o
 ccache gcc -I/usr/local/include/php -I. -I/svr-setup/pecl-memcache-4.0.1-php73 -DPHP_ATOM_INC -I/svr-setup/pecl-memcache-4.0.1-php73/include -I/svr-setup/pecl-memcache-4.0.1-php73/main -I/svr-setup/pecl-memcache-4.0.1-php73 -I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /svr-setup/pecl-memcache-4.0.1-php73/php7/memcache.c  -fPIC -DPIC -o php7/.libs/memcache.o
/svr-setup/pecl-memcache-4.0.1-php73/php7/memcache_pool.c:1647:21: error: conflicting types for ‘mmc_prepare_key_ex’
 MMC_POOL_INLINE int mmc_prepare_key_ex(const char *key, unsigned int key_len, char *result, unsigned int *result_len, char *prefix)  /* {{{ */
                     ^
In file included from /svr-setup/pecl-memcache-4.0.1-php73/php7/memcache_pool.c:40:0:
/svr-setup/pecl-memcache-4.0.1-php73/php7/memcache_pool.h:399:5: note: previous declaration of ‘mmc_prepare_key_ex’ was here
 int mmc_prepare_key_ex(const char *, unsigned int, char *, unsigned int *);
     ^
make: *** [php7/memcache_pool.lo] Error 1
make: *** Waiting for unfinished jobs....
/svr-setup/pecl-memcache-4.0.1-php73/php7/memcache.c: In function ‘php_mmc_store’:
/svr-setup/pecl-memcache-4.0.1-php73/php7/memcache.c:643:4: error: too many arguments to function ‘mmc_prepare_key_ex’
    if (mmc_prepare_key_ex(ZSTR_VAL(key), ZSTR_LEN(key), request->key, &(request->key_len), MEMCACHE_G(key_prefix)) != MMC_OK) {
    ^
In file included from /svr-setup/pecl-memcache-4.0.1-php73/php7/php_memcache.h:34:0,
                 from /svr-setup/pecl-memcache-4.0.1-php73/php7/memcache.c:30:
/svr-setup/pecl-memcache-4.0.1-php73/php7/memcache_pool.h:399:5: note: declared here
 int mmc_prepare_key_ex(const char *, unsigned int, char *, unsigned int *);
     ^
make: *** [php7/memcache.lo] Error 1
```

system 

* CentOS 7.6 64bit
* PHP-FPM 7.3.2
* Centmin Mod 123.09beta01 LEMP stack

PHP 7.3.2 build info

```
php-config
Usage: /usr/local/bin/php-config [OPTION]
Options:
  --prefix            [/usr/local]
  --includes          [-I/usr/local/include/php -I/usr/local/include/php/main -I/usr/local/include/php/TSRM -I/usr/local/include/php/Zend -I/usr/local/include/php/ext -I/usr/local/include/php/ext/date/lib]
  --ldflags           [ -L/usr/local/lib64]
  --libs              [-lcrypt  -lc-client  -lzip -lzip -lz -lexslt -ltidy -largon2 -lresolv -lcrypt -lsodium -ledit -lncurses -laspell -lpspell -lrt -lldap -llber -lstdc++ -lcrypt -lpam -lgmp -lX11 -lXpm -lpng -lz -ljpeg -lwebp -lenchant -lbz2 -lz -lrt -lm -ldl -lnsl  -lsystemd -lxml2 -lz -lm -ldl -lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err -lssl -lcrypto -lcurl -lxml2 -lz -lm -ldl -lssl -lcrypto -lfreetype -lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err -lssl -lcrypto -licui18n -licuuc -licudata -licuio -lxml2 -lz -lm -ldl -lnetsnmp -lssl -lssl -lcrypto -lm -lxml2 -lz -lm -ldl -lcrypt -largon2 -lxml2 -lz -lm -ldl -lxml2 -lz -lm -ldl -lxml2 -lz -lm -ldl -lxml2 -lz -lm -ldl -lxslt -lxml2 -lz -ldl -lm -lssl -lcrypto -lcrypt ]
  --extension-dir     [/usr/local/lib/php/extensions/no-debug-non-zts-20180731]
  --include-dir       [/usr/local/include/php]
  --man-dir           [/usr/local/php/man]
  --php-binary        [/usr/local/bin/php]
  --php-sapis         [ cli embed fpm phpdbg cgi]
  --configure-options [--enable-fpm --enable-opcache --enable-intl --enable-pcntl --with-mcrypt --with-snmp --enable-embed=shared --with-mhash --with-zlib --with-gettext --enable-exif --enable-zip --with-libzip --with-bz2 --enable-soap --enable-sockets --enable-sysvmsg --enable-sysvsem --enable-sysvshm --enable-shmop --with-pear --enable-mbstring --with-openssl --with-mysql=mysqlnd --with-libdir=lib64 --with-mysqli=mysqlnd --with-mysql-sock=/var/lib/mysql/mysql.sock --with-curl --with-gd --with-xmlrpc --enable-bcmath --enable-calendar --enable-ftp --enable-gd-native-ttf --with-freetype-dir=/usr --with-jpeg-dir=/usr --with-png-dir=/usr --with-xpm-dir=/usr --with-webp-dir=/usr --with-t1lib=/usr --enable-pdo --with-pdo-sqlite --with-pdo-mysql=mysqlnd --enable-inline-optimization --with-imap --with-imap-ssl --with-kerberos --with-readline --with-libedit --with-gmp --with-pspell --with-tidy --with-enchant --with-fpm-user=nginx --with-fpm-group=nginx --with-ldap --with-ldap-sasl --with-password-argon2 --with-sodium --with-config-file-scan-dir=/etc/centminmod/php.d --with-fpm-systemd --with-xsl]
  --version           [7.3.2]
  --vernum            [70302]
```

**Comment by centminmod** [comment]
FYI same issue with PHP 7.2.15 as well so reverted to commit https://github.com/websupport-sk/pecl-memcache/commit/4a9e4ab0d12150805feca3012854de9fd4e5a721 which is fine with PHP 7.2.15

**Comment by baer95** [comment]
https://github.com/websupport-sk/pecl-memcache/issues/37 this seems to be the same issue. I am also experiencing the same error - `too many arguments to function ‘mmc_prepare_key_ex’`.

**Comment by faca5** [comment]
Hello.

We have error on 7.3.

/bin/sh /usr/local/src/pecl-memcache-NON_BLOCKING_IO_php72/libtool --mode=compile cc -I/opt/cpanel/ea-php72/root/usr/include/php -I. -I/usr/local/src/pecl-memcache-NON_BLOCKING_IO_php72 -DPHP_ATOM_INC -I/usr/local/src/pecl-memcache-NON_BLOCKING_IO_php72/include -I/usr/local/src/pecl-memcache-NON_BLOCKING_IO_php72/main -I/usr/local/src/pecl-memcache-NON_BLOCKING_IO_php72 -I/opt/cpanel/ea-php72/root/usr/include/php -I/opt/cpanel/ea-php72/root/usr/include/php/main -I/opt/cpanel/ea-php72/root/usr/include/php/TSRM -I/opt/cpanel/ea-php72/root/usr/include/php/Zend -I/opt/cpanel/ea-php72/root/usr/include/php/ext -I/opt/cpanel/ea-php72/root/usr/include/php/ext/date/lib  -DHAVE_CONFIG_H  -g -O2   -c /usr/local/src/pecl-memcache-NON_BLOCKING_IO_php72/php7/memcache.c -o php7/memcache.lo
libtool: compile:  cc -I/opt/cpanel/ea-php72/root/usr/include/php -I. -I/usr/local/src/pecl-memcache-NON_BLOCKING_IO_php72 -DPHP_ATOM_INC -I/usr/local/src/pecl-memcache-NON_BLOCKING_IO_php72/include -I/usr/local/src/pecl-memcache-NON_BLOCKING_IO_php72/main -I/usr/local/src/pecl-memcache-NON_BLOCKING_IO_php72 -I/opt/cpanel/ea-php72/root/usr/include/php -I/opt/cpanel/ea-php72/root/usr/include/php/main -I/opt/cpanel/ea-php72/root/usr/include/php/TSRM -I/opt/cpanel/ea-php72/root/usr/include/php/Zend -I/opt/cpanel/ea-php72/root/usr/include/php/ext -I/opt/cpanel/ea-php72/root/usr/include/php/ext/date/lib -DHAVE_CONFIG_H -g -O2 -c /usr/local/src/pecl-memcache-NON_BLOCKING_IO_php72/php7/memcache.c  -fPIC -DPIC -o php7/.libs/memcache.o
/usr/local/src/pecl-memcache-NON_BLOCKING_IO_php72/php7/memcache.c: In function 'php_mmc_store':
/usr/local/src/pecl-memcache-NON_BLOCKING_IO_php72/php7/memcache.c:643:4: error: too many arguments to function 'mmc_prepare_key_e '
    if (mmc_prepare_key_ex(ZSTR_VAL(key), ZSTR_LEN(key), request->key, &(request->key_len), MEMCACHE_G(key_prefix)) != MMC_OK) {
    ^
In file included from /usr/local/src/pecl-memcache-NON_BLOCKING_IO_php72/php7/php_memcache.h:34:0,
                 from /usr/local/src/pecl-memcache-NON_BLOCKING_IO_php72/php7/memcache.c:30:
/usr/local/src/pecl-memcache-NON_BLOCKING_IO_php72/php7/memcache_pool.h:399:5: note: declared here
 int mmc_prepare_key_ex(const char *, unsigned int, char *, unsigned int *);
     ^
make: *** [php7/memcache.lo] Error 1

**Comment by borodiychuk** [comment]
This is fixed in PR #39
